src/test/regress/sql/create_drop_database_propagation.sql

```diff
@@ -0,0 +1,121 @@
+
+
+\set create_drop_db_tablespace :abs_srcdir '/tmp_check/ts3'
+CREATE TABLESPACE create_drop_db_tablespace LOCATION :'create_drop_db_tablespace';
+
+\c - - - :worker_1_port
+\set create_drop_db_tablespace :abs_srcdir '/tmp_check/ts4'
+CREATE TABLESPACE create_drop_db_tablespace LOCATION :'create_drop_db_tablespace';
+
+\c - - - :worker_2_port
+\set create_drop_db_tablespace :abs_srcdir '/tmp_check/ts5'
+CREATE TABLESPACE create_drop_db_tablespace LOCATION :'create_drop_db_tablespace';
+
+\c - - - :master_port
+create user create_drop_db_test_user;
+
+set citus.enable_create_database_propagation=on;
+
+CREATE DATABASE mydatabase
+    WITH TEMPLATE = 'template0'
+            OWNER = create_drop_db_test_user
+            CONNECTION LIMIT = 10
+            ENCODING = 'UTF8'
+            LC_COLLATE = 'C'
+            LC_CTYPE = 'C'
+            TABLESPACE = create_drop_db_tablespace
+            ALLOW_CONNECTIONS = true
+            IS_TEMPLATE = false;
+
+SELECT pd.datname, pd.encoding,
+pd.datistemplate, pd.datallowconn, pd.datconnlimit,
+pd.datcollate , pd. datctype  ,  pd.datacl,
+pa.rolname AS database_owner, pt.spcname AS tablespace
+FROM pg_database pd
+JOIN pg_authid pa ON pd.datdba = pa.oid
+join pg_tablespace pt on pd.dattablespace = pt.oid
+WHERE datname = 'mydatabase';
```

Comment:
I'd rather run this command via `run_command_on_all_nodes` to avoid from running the same query by individually connecting to other nodes (as done in below).

This is because, it both increases test-completion time and also causes unnecessarily duplicating the test-code.

```sql
SELECT result from run_command_on_all_nodes(
  $$
  SELECT jsonb_agg(to_jsonb(q2.*)) FROM (
    -- your actual query starts here
    SELECT pd.datname, pd.encoding,
    pd.datistemplate, pd.datallowconn, pd.datconnlimit,
    pd.datcollate , pd. datctype  ,  pd.datacl,
    pa.rolname AS database_owner, pt.spcname AS tablespace
    FROM pg_database pd
    JOIN pg_authid pa ON pd.datdba = pa.oid
    join pg_tablespace pt on pd.dattablespace = pt.oid
    WHERE datname = 'mydatabase'
    -- your actual query ends here
  ) q2
  $$
) ORDER BY result;
```

Reply:
(same comment applies to what we do after `drop database` tests and to the tests in pg15.sql)

Reply:
And let's also check pg_dist_object record on all nodes, both after creating / dropping the database:

```sql
SELECT result from run_command_on_all_nodes(
  $$
  SELECT jsonb_agg(to_jsonb(q2.*)) FROM (
    SELECT * FROM (
      SELECT
        pg_identify_object_as_address(classid, objid, objsubid)::text AS obj_def_str,
        type, object_names, object_args, distribution_argument_index, colocationid, force_delegation
      FROM pg_dist_object
    ) q1
    WHERE obj_def_str LIKE '%mydatabase%'
  ) q2
  $$
) ORDER BY result;
```